minimesos/src/main/java/com/containersol/minimesos/cluster/MesosCluster.java

```diff
@@ -108,6 +112,12 @@ private MesosCluster(String clusterId) {
                         this.containers.add(zookeeper);
                         break;
                     case "agent":
+                        File slaveMappingDir = new File(MesosCluster.getHostDir(), ".minimesos/sandbox-" + clusterId + "/" + SLAVE_MAPPING_DIR + "-" + uuid);
+                        try {
+                            FileUtils.forceMkdir(slaveMappingDir);
+                        } catch (IOException e) {
+                            // ignore
+                        }
```

Comment:
Extract method `deserializeCluster` and wrap this method in a try catch block. If a cluster cannot be deserialized properly it should log this and remove any left over containers.